doc/architecture/frontend.md

```diff
@@ -0,0 +1,87 @@
+- `config/`: Contains initialization code for the environment and libraries.
+- `features/`: Contains the distinct features of the application.
+- `generated/`: Contains generated code, for now only docs.
+- `hooks/`: Contains hooks that are shared across features.
+- `icons/`: Contains svg icons.
+- `routes/`: Contains the routes of the application.
+- `styles/`: Contains the global and other style sheets, e.g. to customize a component from a third party library like `react-leaflet`.
+- `utils/`: Contains additional utility functions, that are shared across features or components and are not hooks.
+
+```mermaid
+flowchart LR
+  routes --> |Renders route features| features
+  features -->|Uses bindings for API calls| bindings
+  features -->|Uses shared components| components
+  features -->|Calls functions| config
+  features -->|Uses hooks| hooks
+  features -->|Calls functions| utils
+  features -->|Uses icon files| icons
+  features -->|Uses asset files| assets
+  features -->|Uses some miscellaneous styles| styles
+  features ---->|Uses another feature's public API| i
+    subgraph f[example feature]
+      direction TB
+      i[index]
+      e[fa:fa-folder example feature structure] --> |Components scoped to a specific feature| c[components]
+      e -->|API request declarations and API hooks related to a specific feature| a[API]
+      e -->|Route components for a specific feature's pages| r[routes]
+      e -->|State stores for a specific feature| s[stores]
+    end
+```
+
+`__mocks__` and `generated` are not used during runtime
+
+### \_\_mocks\_\_
+
+This folder has a special name to be recognized by our test runner jest.
+More details on how mocking with jest works can be found in the [jest documentation](https://jestjs.io/docs/manual-mocks).
+
+### assets
+
+This folder should contain only small asset files like SVGs, so that our bundle size stays small.
+Larger files should be hosted on Nextcloud.
+
+### bindings
+
+This folder contains API bindings that are generated by `typeshare`.
+They can be regenerated by running the command `npm run generate-api-types`.
+
+### components
+
+All components that reside inside here need to be documented in storybook.
+
+The minimum requirements for documentation are:
+
+- If the component can be in different states (e.g. a visual variant, or an open/closed state), each state should be described by a story.
+- Every property that can be passed to the component needs to be documented.
+
+### features
+
+Most of the application's code lives here.
+Every folder created here represents a given feature and contains its domain specific code.
+For example, if a feature interacts with the backend via network requests, it would have a sub module `api` that encapsulates this.
+
+Features are allowed to import another features public API which is exported from its `index.ts` file.
+Features are also allowed to import all other previously mentioned modules if needed.
+
+### Code Documentation
```

Comment:
```suggestion
### code documentation
```

Reply:
@absurd-turtle the original capitalization is correct

Reply:
I suggested it to be consistent with the other headlines but maybe the reason that they are not the same is that they represent the directories while this one does not.

Reply:
I think the other headlines are folder names, and in such special situations (as it is case-sensitive) it is okay. Maybe they should be written like this (or see other comment):

```
##  `assets`
```